core/templates/pages/admin-page/roles-tab/admin-roles-tab.directive.ts

```diff
@@ -65,48 +90,239 @@ angular.module('oppia').directive('adminRolesTab', [
+          AdminTaskManagerService.finishTask();
+        };
+
+        ctrl.submitViewCommunityReviewersForm = function(formResponse) {
+          if (AdminTaskManagerService.isTaskRunning()) {
+            return;
+          }
+          ctrl.setStatusMessage('Processing query...');
+          AdminTaskManagerService.startTask();
+          if (formResponse.method === VIEW_METHOD_ROLE) {
+            $http.get(
+              '/getcommunityreviewershandler', {
+                params: {
+                  type: formResponse.type,
+                  language_code: formResponse.languageCode
+                }
+              }).then(function(response) {
+              ctrl.result.usernames = response.data.usernames;
+              ctrl.communityReviewersDataFetched = true;
+              ctrl.setStatusMessage('Success.');
+            }, handleErrorResponse);
+          } else {
+            var translationLanguages = [];
+            var voiceoverLanguages = [];
+            $http.get(
+              '/communityreviewerrightsdatahandler', {
+                params: {
+                  username: formResponse.username
+                }
+              }).then(function(response) {
+              translationLanguages = getLanguageDescriptions(
+                response.data.can_review_translation_for_language_codes);
+              voiceoverLanguages = getLanguageDescriptions(
+                response.data.can_review_voiceover_for_language_codes);
+              ctrl.result = {
+                translationLanguages: translationLanguages,
+                voiceoverLanguages: voiceoverLanguages,
+                questions: response.data.can_review_questions
+              };
+              ctrl.communityReviewersDataFetched = true;
+              ctrl.setStatusMessage('Success.');
+            }, handleErrorResponse);
+          }
           AdminTaskManagerService.finishTask();
         };
+
+        ctrl.submitRemoveCommunityReviewerForm = function(formResponse) {
+          if (AdminTaskManagerService.isTaskRunning()) {
+            return;
+          }
+          ctrl.setStatusMessage('Processing query...');
+          AdminTaskManagerService.startTask();
+          $http.put(
+            '/removecommunityreviewerhandler', {
+              username: formResponse.username,
+              removal_type: formResponse.method,
+              review_type: formResponse.type,
+              language_code: formResponse.languageCode
+            }).then(function(response) {
+            ctrl.setStatusMessage('Success.');
+            refreshFormData();
+          }, handleErrorResponse);
+          AdminTaskManagerService.finishTask();
+        };
+
+        var refreshFormData = function() {
+          ctrl.formData = {
+            viewUserRoles: {
+              method: VIEW_METHOD_ROLE,
+              role: null,
+              username: '',
+              isValid: function() {
+                if (this.method === VIEW_METHOD_ROLE) {
```

Comment:
> Perhaps change the name to "filter criterion" or similar

>> I tried using the existing terminology. Should I change it?

Hmm. I think yes, in this case, since it should add clarity.